guide/event-driven-and-pooling-consumer-service/tests/event-driven-and-pooling-consumer_test.bal

```diff
@@ -0,0 +1,173 @@
+endpoint mb:SimpleQueueSender queueHealthCheck {
+    host: "localhost",
+    port: 5672,
+    queueName: "SensorHealth"
+};
+documentation { Define the message queue endpoint for maintenance messages. }
+endpoint mb:SimpleQueueSender queueMaintenance {
+    host: "localhost",
+    port: 5672,
+    queueName: "Maintenance"
+};
+documentation { Define the message queue endpoint for sensor calibration requests. }
+endpoint mb:SimpleQueueSender queueCalibration {
+    host: "localhost",
+    port: 5672,
+    queueName: "Calibration"
+};
+documentation { Attributes associated with the service endpoint. }
+endpoint http:Listener sensorEventListner {
+    port: 8080
+};
+documentation {  via HTTP/1.1. }
+@http:ServiceConfig {
+    basePath: "/"
+}
+service<http:Service> SensorEventService bind sensorEventListner {
+    @http:ResourceConfig {
+        methods: ["POST"], consumes: ["application/json"], produces: ["application/json"],
+        path: "/activity"
+    }
+    documentation { Resources for reciving geo activities captured from the sensor }
+    activity(endpoint conn, http:Request req) {
+        json requestMessage = check req.getJsonPayload();
+        requestmessage1 = untaint requestMessage.toString();
+    }
+    @http:ResourceConfig {
+        methods: ["POST"],
+        path: "/health"
+    }
+    documentation { Resource for checking the health of the EP of the sensor}
+    health(endpoint conn, http:Request req) {
+        http:Response res = new;
+        json requestMessage = check req.getJsonPayload();
+        requestmessage2 = untaint requestMessage.toString();
+    }
+    @http:ResourceConfig {
+        methods: ["POST"],
+        path: "/maintenance"
+    }
+    documentation { Resource for invoking a service when the sensor needed mainteinance }
+    maintanance(endpoint conn, http:Request req) {
+        http:Response res = new;
+        json requestMessage = check req.getJsonPayload();
+        requestmessage3 = untaint requestMessage.toString();
+        res.setPayload("Ack - maintenance team on the way");
+    }
+    @http:ResourceConfig {
+        methods: ["POST"],
+        path: "/calibrate"
+    }
+    documentation { Resource for invoking a service when the sensor needed calibration }
+    calibrate(endpoint conn, http:Request req) {
+        http:Response res = new;
+        json requestMessage = check req.getJsonPayload();
+        requestmessage4 = untaint requestMessage.toString();
+        res.setPayload("Ack - Calibration scheduled");
+    }
+}
+function messageSender() {
+    http:Request req1;
+    json sampleRequest1 = { "ID": 27125088, "SID": 4344, "TIME_S": "1536423224", "TIME_E": "1536423584", "TYPE": "EQ",
+        "C_DATA": "41.40338, 2.17403", "R_SCALE": 10 };
+    string msg1 = sampleRequest1.toString();
+    mb:Message message1 = check queuesensorGeoMessage.createTextMessage(msg1);
+    _ = queuesensorGeoMessage->send(message1);
```

Comment:
Better to handle the errors rather than ignoring.

Reply:
Add errors handling whenever possible.